client/my-sites/earn/ads/form-settings.jsx

```diff
@@ -17,76 +16,103 @@ import FormTextInput from 'calypso/components/forms/form-text-input';
 import FormTextarea from 'calypso/components/forms/form-textarea';
 import SectionHeader from 'calypso/components/section-header';
 import SupportInfo from 'calypso/components/support-info';
-import { protectForm } from 'calypso/lib/protect-form';
+import { ProtectFormGuard } from 'calypso/lib/protect-form';
+import { useDispatch, useSelector } from 'calypso/state';
 import getSiteUrl from 'calypso/state/selectors/get-site-url';
 import { getWordadsSettings } from 'calypso/state/selectors/get-wordads-settings';
 import isSavingWordadsSettings from 'calypso/state/selectors/is-saving-wordads-settings';
 import { isJetpackSite, getCustomizerUrl } from 'calypso/state/sites/selectors';
 import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
-import { dismissWordAdsSuccess } from 'calypso/state/wordads/approve/actions';
 import { saveWordadsSettings } from 'calypso/state/wordads/settings/actions';
 
-class AdsFormSettings extends Component {
-	static propTypes = {
-		site: PropTypes.object.isRequired,
-		markChanged: PropTypes.func.isRequired,
-		markSaved: PropTypes.func.isRequired,
-	};
-
-	state = {};
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( { wordadsSettings, site } ) {
```

Comment:
There a wider project to remove UNSAFE life cycle methods, which this PR does. I've updated the tracking issue for that issue [here](https://github.com/Automattic/wp-calypso/issues/58453).